addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -0,0 +1,332 @@
+                'product_uom_qty': 1,
+                'product_uom': product.uom_id.id,
+                'order_id': self.id,
+                'price_unit': 0,  # computed by _cart_adjust_rewards
+                'tax_id': [fields.Command.link(tax.id) for tax in taxes],
+                'sequence': REWARD_SALE_ORDER_LINE_SEQUENCE,
+            })
+        self._cart_adjust_rewards()
+        website = self.env['website'].get_current_website()
+        self.recompute_loyalty_points(website.loyalty_id.id)
+
+    def _get_won_points(self, loyalty):
+        """The total of points won, excluding the points spent on rewards"""
+        if self.website_id.is_public_user() or not loyalty:
+            return 0
+        total_points = 0
+        for line in self.order_line.filtered(lambda line: not line.loyalty_reward_id):
+            line_points = 0
+            for rule in loyalty.rule_ids:
+                rule_points = 0
+                if rule.is_product_valid(line.product_id):
+                    rule_points += rule.points_quantity * line.product_uom_qty
+                    rule_points += rule.points_currency * line.price_total
+                if rule_points > line_points:
+                    line_points = rule_points
+
+            total_points += line_points
+
+        total_points += self.amount_total * loyalty.points
+        return max(0, tools.float_round(total_points, 0, rounding_method='HALF-UP'))
+
+    def _get_spent_points(self):
+        """The total number of points spent on rewards"""
+        if not self.partner_id:
+            return 0
+        points = 0
+        for line in self.order_line.filtered(lambda line: line.loyalty_reward_id and line.is_main_loyalty_reward):
+            line_points = line.product_uom_qty * line.loyalty_reward_id.point_cost
+            points += tools.float_round(line_points, 0, rounding_method='HALF-UP')
+        return max(0, points)
+
+    def recompute_loyalty_points(self, loyalty_id):
+        loyalty = self.env['loyalty.program'].browse(loyalty_id) if loyalty_id else None
+        for order in self:
+            if loyalty:
+                order.won_loyalty_points = order._get_won_points(loyalty)
+                order.spent_loyalty_points = order._get_spent_points()
+            else:
+                order.won_loyalty_points = 0
+                order.spent_loyalty_points = 0
+
+    def get_portal_loyalty_url(self):
+        return self.get_portal_url().replace('orders', 'loyalty')
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    loyalty_reward_id = fields.Many2one('loyalty.reward', string='Loyalty Reward', help='The loyalty reward of this line')
+    is_main_loyalty_reward = fields.Boolean(string='Main Loyalty Reward', help='True only for the main line of a loyalty reward')
+
+    def _is_not_sellable_line(self):
+        return self.loyalty_reward_id and self.loyalty_reward_id.reward_type == 'discount' or super()._is_not_sellable_line()
```

Comment:
If `self.loyalty_reward_id` is falsy, then `self.loyalty_reward_id.reward_type` returns `False`, so the first check is useless:
```suggestion
        return self.loyalty_reward_id.reward_type == 'discount' or super()._is_not_sellable_line()
```